packages/radix-ui-themes/src/components/theme-panel.tsx

```diff
@@ -23,605 +24,628 @@ import { themePropDefs } from '../props/index.js';
 import type { ComponentPropsWithout, RemovedProps } from '../helpers/index.js';
 import type { GetPropDefTypes } from '../props/index.js';
 
-interface ThemePanelProps extends Omit<ThemePanelImplProps, keyof ThemePanelImplPrivateProps> {
+const keyboardInputElement = `
+  [contenteditable],
+  [role="combobox"],
+  [role="listbox"],
+  [role="menu"],
+  input:not([type="radio"], [type="checkbox"]),
+  select,
+  textarea
+`;
+
+/** Listen to keydown events and fire a callback when a hotkey is pressed */
+const useHotKey = (
+  key: string | null | false,
+  callback: () => void,
+  options?: {
+    /** Determines whether the hotkey listener is active or not */
+    enabled?: boolean;
+  }
+) => {
+  const { enabled = true } = options ?? {};
+  const callbackRef = useCallbackRef(callback);
+  React.useEffect(() => {
+    if (!key) return;
+    if (enabled === false) return;
+    function handleKeydown(event: KeyboardEvent) {
+      const isModifierActive = event.altKey || event.ctrlKey || event.shiftKey || event.metaKey;
+      const isKeyboardInputActive = document.activeElement?.closest(keyboardInputElement);
+      const isKeyActive = event.key?.toUpperCase() === (key as string).toUpperCase();
+      if (isKeyActive && !isModifierActive && !isKeyboardInputActive) {
+        callbackRef();
+      }
+    }
+    document.addEventListener('keydown', handleKeydown);
+    return () => document.removeEventListener('keydown', handleKeydown);
+  }, [key, callbackRef, enabled]);
+};
+
+type ThemePanelElement = React.ElementRef<'div'>;
+
+interface ThemePanelProps extends ComponentPropsWithout<'div', RemovedProps> {
+  open?: boolean;
+  onOpenChange?: (open: boolean) => void;
+  /** Whether the theme panel is open by default.
+   * Doesn't have any effect if `open` is also set.
+   * @default true
+   */
   defaultOpen?: boolean;
+  onAppearanceChange?: (value: 'light' | 'dark') => void;
+  /** A hotkey to quickly show/hide the panel.
+   * Set to `null` or `false` to disable the hotkey.
+   * @default "T"
+   */
+  openHotkey?: string | null | false;
+  /** A hotkey to quickly toggle the appearance.
+   * Set to `null` or `false` to disable the hotkey.
+   * @default "D"
+   * */
+  toogleAppearanceHotkey?: string | null | false;
```

Comment:
We wouldn't want to expose a way to customize the shortcut itself – the theme panel should always work the same in different places. Having different shortcuts on different websites is going to feel unpredictable.

Just a boolean prop to disable the shortcuts should be fine, e.g. `disableShortcuts`

Reply:
I respectfully disagree with your perspective. Default behavior can indeed remain consistent, but offering customization options allows users to adapt the component to fit seamlessly into their workflows. If a user chooses to change a defined hotkey, it should be within their responsibility, and providing such flexibility enhances the usability and adaptability of the library.

Reply:
By the way, just noticed that Radix Primitives has an example of [Custom hotkey](https://www.radix-ui.com/primitives/docs/components/toast#custom-hotkey) prop for Toast component 🙂

![image](https://github.com/radix-ui/themes/assets/15171036/9c2ed184-5567-4d39-be07-fed35b5d09b5)

Reply:
FWIW Radix Primitives is designed to be more flexible, where Themes is a more opinionated starter. I could be convinced on this, however. I agree it can be annoying to deal with potential conflicts.